### DISCUSSION TITLE
Welcome to zfltk Discussions!

### DISCUSSION THREAD

**Discussion by MoAlyousef** [MoAlyousef/zfltk#14] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by jcbritobr** [comment]
Hello.
My name is Julio Gardona and I'm a Brazilian software engineer working at Embraer. I worked in telecomunications area for many years, and today I work with airplanes.

Nice work as always you did with fltk library. Thank you for your work with fltk.

I would like to ask what advantages zig can bring over rust, about using fltk as wrapper library. 
Its safer? Better performance or easier to use and code?

**Reply by mgord9518** [reply]
Zig is much simpler syntactically than Rust but also doesn't have as many features, so it's mostly personal preference on whether it's "better" or not. I think Zig tends to lean more towards the philosophy of extending existing projects instead of completely rewriting them, which is common to see in Rust.

Zig is a bit less safe than Rust, but is MUCH, MUCH safer than C. While it doesn't have a borrow checker, it does have non-nullable pointers, bounds checking, a testing allocator that automatically detects memory leaks, a great stdlib and pretty much unmatched interop with C.

Ultimately, they're both amazing, performant languages and the only way to know which will be better for you would be to mess with both of them. Something to look out for is that Zig is still a young language and isn't stable, so if you do decide to write a project in it, you'll likely have to update it as the language evolves.

**Reply by MoAlyousef** [reply]
Welcome Julio. Thank you for the kind words.
In addition to what @mgord9518 said, Zig has easier C interop. It's also easier to use as a drop-in replacement, since you don't need to install tools like rust-bindgen, you can include a C header directly and link to the C library directly. 
Also not all software needs the memory and thread safety brought by Rust (at the cost of language-complexity), especially for non-network facing programs. 
That said, Zig is still at pre-1.0. so things are breaking left and right. It also lacks an official package manager (still being worked on), so adding a package is still a mess (third party package managers which don't always work and using git submodules or vendoring deps).

**Reply by jcbritobr** [reply]
I did some zig programming months ago and the language looks very promising. Its very simplistic like golang. Have fltk working on zig seems a very interesting project as we can use to build tools in faster way for production. Unfortunately zig is not done yet, but I will like to use it in future for production.
